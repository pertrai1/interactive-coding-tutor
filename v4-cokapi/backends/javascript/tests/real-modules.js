// Test real module syntax that should be transformed

// Named imports  
import { add, subtract } from './math';

// Default import
import Calculator from './calculator';

// Named exports
export function multiply(a, b) {
  return a * b;
}

// Default export
export default function divide(a, b) {
  return a / b;
}

// Use the imported functions
var sum = add(5, 3);
var calc = new Calculator();