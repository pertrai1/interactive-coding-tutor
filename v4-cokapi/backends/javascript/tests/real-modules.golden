{"code":"// Test real module syntax that should be transformed\n\n// Named imports  \nimport { add, subtract } from './math';\n\n// Default import\nimport Calculator from './calculator';\n\n// Named exports\nexport function multiply(a, b) {\n  return a * b;\n}\n\n// Default export\nexport default function divide(a, b) {\n  return a / b;\n}\n\n// Use the imported functions\nvar sum = add(5, 3);\nvar calc = new Calculator();","trace":[{"line":6,"event":"step_line","func_name":"<module>","globals":{},"ordered_globals":[],"stack_to_render":[],"heap":{},"stdout":""},{"line":7,"event":"step_line","func_name":"<module>","globals":{},"ordered_globals":[],"stack_to_render":[],"heap":{},"stdout":""},{"line":9,"event":"step_line","func_name":"<module>","globals":{},"ordered_globals":[],"stack_to_render":[],"heap":{},"stdout":""},{"line":12,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"]},"ordered_globals":["_math"],"stack_to_render":[],"heap":{},"stdout":""},{"line":15,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"],"_calculator":["INSTANCE","Object"]},"ordered_globals":["_math","_calculator"],"stack_to_render":[],"heap":{},"stdout":""},{"line":19,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"],"_calculator":["INSTANCE","Object"]},"ordered_globals":["_math","_calculator"],"stack_to_render":[],"heap":{},"stdout":""},{"line":23,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"],"_calculator":["INSTANCE","Object"]},"ordered_globals":["_math","_calculator"],"stack_to_render":[],"heap":{},"stdout":""},{"line":28,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"],"_calculator":["INSTANCE","Object"],"sum":8},"ordered_globals":["_math","_calculator","sum"],"stack_to_render":[],"heap":{},"stdout":""},{"line":29,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"],"_calculator":["INSTANCE","Object"],"sum":8,"calc":["INSTANCE","Calculator"]},"ordered_globals":["_math","_calculator","sum","calc"],"stack_to_render":[],"heap":{},"stdout":""}]}
